pkg/validation/validation.go

```diff
@@ -0,0 +1,220 @@
+/*
+
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/nov1n/kubernetes-workflow/pkg/api"
+
+	k8sApi "k8s.io/kubernetes/pkg/api"
+	k8sValidationUnv "k8s.io/kubernetes/pkg/api/unversioned/validation"
+	k8sValidation "k8s.io/kubernetes/pkg/api/validation"
+	k8sSets "k8s.io/kubernetes/pkg/util/sets"
+	k8sField "k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+// ValidateWorkflow validates a workflow
+func ValidateWorkflow(workflow *api.Workflow) k8sField.ErrorList {
+	// Workflows and rcs have the same name validation
+	allErrs := k8sValidation.ValidateObjectMeta(&workflow.ObjectMeta, true, k8sValidation.ValidateReplicationControllerName, k8sField.NewPath("metadata"))
```

Comment:
What if k8sValidation.ValidateReplicationControllerName changes?
We probably want to implement our own or just use k8sValidation.NameIsDNSSubdomain which is also used by k8sValidation.ValidateReplicationControllerName (see image).
![validation_-_godoc_](https://cloud.githubusercontent.com/assets/5707216/15425695/5d6f04bc-1e8a-11e6-94ed-0b1c7dd086b0.png)